pkg/storage/ingest/reader.go

```diff
@@ -118,8 +121,9 @@ func (noopPusherCloser) Close() []error {
 }
 
 func NewPartitionReaderForPusher(kafkaCfg KafkaConfig, partitionID int32, instanceID string, pusher Pusher, logger log.Logger, reg prometheus.Registerer) (*PartitionReader, error) {
+	metrics := newPusherConsumerMetrics(reg)
```

Comment:
What's the reasoning for this change? I'd rather let the constructor of `newPusherConsumer` create the dependency it needs. I don't see `metrics` being used anywhere else.

Reply:
Right! Great catch.